architecture/MIGRATIONS.MD

```diff
@@ -24,3 +24,54 @@ $ date '+%Y%m%d%H%M%S'
 ### Transactional Migrations
 
 Each migration file will run in a single transaction, which means that you can not have for example multiple `CREATE INDEX CONCURRENTLY` statements in a given migration file. Instead break each statement out into it's own migration file.
+
+## Troubleshooting
+
+### Migration Table Locks
+
+Bun (the db client) uses a simple locking mechanism while running migrations that locks an entire table by attempting to write a row `(id, table_name)` to a table called `bun_migration_locks`. The table uses `table_name` as a unique key so that only one process can lock a table at a time. After the process finishes modifying the table, it deletes its lock.
+
+However, a problem occurs if the process dies while still holding a lock, as the lock will never be deleted & no other process will be allowed to lock the table.
```

Comment:
yeah, on tesetnet. potentially because of my own rapid fire deploys 😄
noticed the migrations weren't running. turned on db query logging and noticed the service continually getting blocked on lock creation 🥴